src/resources/resources.service.ts

```diff
@@ -2,43 +2,39 @@ import {
     BadRequestException,
     Injectable,
     NotFoundException,
-    UnauthorizedException,
 } from "@nestjs/common";
 import { CreateResourceDto } from "./dto/create-resource.dto";
 import { UpdateResourceDto } from "./dto/update-resource.dto";
 import { PrismaService } from "../prisma/prisma.service";
-import { GlobalService } from "../global/global.service";
+import { CustomRequest } from "../global/types/CustomRequest";
+import { manageOwnVoyageTeamWithIdParam } from "../ability/conditions/voyage-teams.ability";
+import { manageOwnResourceById } from "../ability/conditions/resource.ability";
 
 @Injectable()
 export class ResourcesService {
-    constructor(
-        private prisma: PrismaService,
-        private readonly globalService: GlobalService,
-    ) {}
+    constructor(private prisma: PrismaService) {}
 
     async createNewResource(
-        req,
+        req: CustomRequest,
         createResourceDto: CreateResourceDto,
         teamId: number,
     ) {
-        const { url, title } = createResourceDto;
-        const userId = req.user.userId;
-
         // make sure teamId exists
         await this.checkTeamExists(teamId);
 
-        const teamMember =
-            await this.globalService.validateLoggedInAndTeamMember(
```

Comment:
I like that you're replacing this as I plan to retire this in future as it requires us to query the database when we already have the information in `req.user`